src/components/sidebar/Sidebar.tsx

```diff
@@ -1,56 +1,116 @@
 import React from 'react'
-
+import { useState } from 'react'
 import * as FaIcons from 'react-icons/fa'
 
+import Home from './Home'
+
 export const SidebarData = [
   {
-    title: 'Home',
-    path: '/',
+    name: 'home',
     icon: <FaIcons.FaHome />,
   },
   {
-    path: '/tasks',
+    name: 'search',
     icon: <FaIcons.FaSearch />,
   },
   {
-    path: '/pen',
+    name: 'pen',
     icon: <FaIcons.FaPen />,
   },
   {
-    path: '/citations',
+    name: 'citations',
     icon: <FaIcons.FaFileExport />,
   },
   {
-    path: '/share',
+    name: 'share',
     icon: <FaIcons.FaShareAlt />,
   },
   {
-    path: '/info',
+    name: 'info',
     icon: <FaIcons.FaInfoCircle />,
   },
   {
-    path: '/download',
+    name: 'download',
     icon: <FaIcons.FaDownload />,
   },
 ]
 
 const Sidebar: React.FunctionComponent = () => {
+  const [homeIconClicked, setHomeIconClicked] = useState(false)
```

Comment:
This is unnecessary, you could write something like:

```ts
const [activeSidebar, setActiveSidebar] = useState<Sidebars>(null)
```


`ISidebars` can be defined like this:
```ts
enum Sidebars {
  SEARCH: 'SEARCH',
  SHARE: 'SHARE',
  // etc...
}
```
And each button will have an onclick handler to change active sidebar:
```ts
export const SideBarData = [
  {
    name: 'search',
    icon: <SomeIcon/>
    onClick: () => setActiveSidebar(Sidebars.SEARCH)
  },
  // etc...
]
```